pkg/sql/pgwire/write_buffer.go

```diff
@@ -96,41 +95,17 @@ func (b *writeBuffer) writeFromFmtCtx(fmtCtx *tree.FmtCtx) {
 	}
 }
 
-// writePaddedData checks if the data needs to be padded and calls
-// applyPadding. Else it writes the data to the write buffer.
-func (b *writeBuffer) writePaddedData(v []byte, t *types.T) {
-	// Add t.Width() as length prefix.
-	b.writeLengthPrefixedByteSlice(v, t.Width())
-
-	remainingPadLength := int(t.Width()) - len(v)
-	b.applyPadding(remainingPadLength)
-}
-
-// applyPadding applies padding (in the form of blanks spaces)
-// to the right of the values inside the write buffer
-// to fill all the remaining width from t.Width().
-func (b *writeBuffer) applyPadding(remainingPadLength int) {
-	for remainingPadLength > 0 {
-		padChunkLength := min(remainingPadLength, len(spaces))
-
-		padChunk := spaces[:padChunkLength]
-		b.write(padChunk)
-
-		remainingPadLength -= padChunkLength
-	}
-}
-
 // writeLengthPrefixedString writes a length-prefixed string. The
 // length is encoded as an int32.
-func (b *writeBuffer) writeLengthPrefixedString(s string) {
-	b.putInt32(int32(len(s)))
+func (b *writeBuffer) writeLengthPrefixedString(s string, specifiedLength int) {
```

Comment:
Let's leave this function as is. See my suggested changes here: https://github.com/cockroachdb/cockroach/commit/c8d771da5b6c6bddc7200f41e987e40a14a25f16